projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp

```diff
@@ -0,0 +1,232 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include <transport/adi_wifi_transport.h>
+#include <transport/adi_uart.h>
+
+using namespace adi_sensor_swpack;
+
+ADI_UART_HANDLE  phDevice;
+uint8_t         aDeviceMemory[ADI_UART_BIDIR_MEMORY_SIZE];
+
+uint8_t Buffer[12];
+
+uint8_t u8Index;
+bool boCommandReceived = false;;
+
+/**
+  @brief UART interrupt callback.
+
+  @param pCBParam - Client supplied callback param.
+  @param Event - Event ID specific to the Driver/Service.
+  @param pArg - parameters list.
+
+  @return void.
+
+**/
+void UARTCallback(
+		void      *pCBParam,         /*!< Client supplied callback param */
+		uint32_t   Event,            /*!< Event ID specific to the Driver/Service */
+		void      *pArg)
+
+{
+	uint32_t             *pHwError;
+
+	if (Event == ADI_UART_EVENT_RX_DATA)
+	{
+		adi_uart_ReadChar(phDevice, &Buffer[u8Index], pHwError);
+		u8Index++;
+	}
+
+	if ((Buffer[u8Index - 1] == '\r') | (u8Index == 12))
+	{
+		boCommandReceived = true;
+		u8Index = 0;
+	}
+}
+
+/**
+  @brief Function to process commands received over UART.
+
+  @param u8Channel - DAC channel to setup.
+  @param u16DacValue - value to set on DAC output.
+
+  @return 0 - Succes, -1 - Failure.
+
+**/
+uint8_t ProcessCommand(uint8_t * u8Channel, uint16_t * u16DacValue)
+{
+	uint16_t u16RawValue;
+
+	if (((Buffer[4] != 'a') &  (Buffer[4] != 'b') & (Buffer[4] != 'c') & (Buffer[4] != 'z')))
+	{
+		return -1;
+	}
+	else
+	{
+
+		if ((Buffer[4] != 'z'))
+		{
+			*u8Channel = Buffer[4]; //channel letter
+		}
+		else
+		{
+			*u8Channel = '0';
+		}
+
+		u16RawValue = (uint16_t)atoi((char *)&Buffer[6]);
+		*u16DacValue = u16RawValue;//(uint32_t)(u16RawValue * 2)/2048 * 65535;
+		return 0;
+	}
+}
+
+int main(int argc, char *argv[])
+{
+	CN0410 cn0410;
+
+	uint8_t u8Channel;
+	uint16_t  u16DacValue, u16ChannelValue;
+	uint8_t u8Error;
+
+	/**
+	 * Initialize managed drivers and/or services that have been added to 
+	 * the project.
+	 * @return zero on success 
+	 */
+	adi_initComponents();
+	
+	timer_start(); //init timer
+
+	adi_uart_Open(ADI_UART_DEVICE_NUM_CFG, ADI_TAL_UART_DIRECTION, aDeviceMemory, ADI_UART_BIDIR_MEMORY_SIZE,  &phDevice);
+	adi_uart_RegisterCallback(phDevice, UARTCallback);
+	adi_uart_SetConfiguration(phDevice, ADI_UART_NO_PARITY, ADI_UART_ONE_STOPBIT, ADI_UART_WORDLEN_8BITS);
+	adi_uart_ConfigBaudRate(phDevice, UART_DIV_C, UART_DIV_M, UART_DIV_N, UART_OSR);
+
+	cn0410.Init(); //init spi and pins
+
+	cn0410.SendCommand(AD5686_ITERNAL_REFERENCE, 0x00, 0x00); //enable internal reference
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+
+	timer_sleep(10); //10 ms delay
+
+	cn0410.SendCommand(AD5686_RESET, 0x00, 0x00); //soft reset to zero scale
```

Comment:
please check if this RESET command doesn't overwrite the values that you have configured before for internal reference and power mode